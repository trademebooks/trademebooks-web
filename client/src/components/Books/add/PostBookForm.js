import React, { Component } from 'react';
import { connect } from 'react-redux';

import './PostBookForm.scss';

class PostBookForm extends Component {
    constructor(props) {
        super(props);

        this.state = {
            image: "",
            title: "",
            authors: [""],
            edition: "",
            isbn10: "",
            isbn13: "",
            condition: "",

            editable: true,

        };

        this.onChangeBook = this.onChangeBook.bind(this);
        this.toggleEditable = this.toggleEditable.bind(this);
        this.selectCondition = this.selectCondition.bind(this);
        this.addAuthor = this.addAuthor.bind(this);
        this.onChangeAuthor = this.onChangeAuthor.bind(this);
        this.removeAuthor = this.removeAuthor.bind(this);
    }

    onChangeBook(e) {
        this.setState({
            [e.target.name]: e.target.value
        });
    }

    toggleEditable(e) {
        this.setState({
            editable: !this.state.editable
        });
    }

    selectCondition(cond) {
        this.setState({
            condition: cond
        });
    }

    addAuthor(e) {
        e.preventDefault();
        this.setState({
            authors: [...this.state.authors, ""]
        });
    }

    removeAuthor(i) {
        let dupe = Array.from(this.state.authors);
        dupe.splice(i, 1);
        this.setState({
            authors: dupe
        });
    }

    onChangeAuthor(i, event) {
        let authors = this.state.authors.slice();
        authors[i] = event.target.value;
        this.setState({ authors: authors });
    }


    render() {
        let authors = this.state.authors.map((author, i) => {
            let currBtn = i !== 0 ?
                <button type="button" className="add-remove-author" onClick={() => this.removeAuthor(i)}>X</button> :
                <button type="button" className="add-remove-author" onClick={this.addAuthor}>Add Another Author</button>;

            return <tr>
                <td>{i !== 0 ? '' : 'Author(s):'}</td>
                <td>
                    <input type="text" name="author" value={author} onChange={(e) => this.onChangeAuthor(i, e)} />

                    {currBtn}
                </td>
            </tr>
        });

        let form;
        if (this.state.editable) {
            form =
                <table>
                    <tbody>
                        <tr>
                            <td>Title:</td>
                            <td><input type="text" name="title" onChange={this.onChangeBook} value={this.state.title}
                                className="full-width" /></td>
                        </tr>
                        {authors}

                        <tr>
                            <td>ISBN-10:</td>
                            <td><input type="text" name="isbn10" onChange={this.onChangeBook} value={this.state.isbn10}
                                className="full-width" /></td>
                        </tr>
                        <tr>
                            <td>ISBN-13:</td>
                            <td><input type="text" name="isbn13" onChange={this.onChangeBook} value={this.state.isbn13}
                                className="full-width" /></td>
                        </tr>
                        <tr>
                            <td>Edition:</td>
                            <td><input type="text" name="edition" onChange={this.onChangeBook} value={this.state.edition}
                                className="full-width" /></td>
                        </tr>
                    </tbody>
                </table>
        } else {
            form =
                <table>
                    <tbody>
                        <tr>
                            <td>Title:</td>
                            <td><p type="text" className="book-title" name="title">{this.state.title}</p></td>
                        </tr>
                        <tr>
                            <td>Author(s):</td>
                            <td><p type="text" className="book-author" name="author">{this.state.authors}</p></td>
                        </tr>
                        <tr>
                            <td>ISBN-10:</td>
                            <td><p type="text" className="isbn10" name="isbn10">{this.state.isbn10}</p></td>
                        </tr>
                        <tr>
                            <td>ISBN-13:</td>
                            <td><p type="text" className="isbn13" name="isbn13">{this.state.isbn13}</p></td>
                        </tr>
                        <tr>
                            <td>Edition:</td>
                            <td><p type="text" className="edition" name="edition">{this.state.edition}</p></td>
                        </tr>
                    </tbody>
                </table>

        }

        let editOpt = !this.state.editable ?
            <p>Information does not look correct? <span onClick={this.toggleEditable}>Manually Edit</span></p> :
            <span></span>

        let conditions = [
            {
                name: "Poor",
                img: require('../../../img/condition_icons/Poor_condition_default.png'),
                desc: "No damage, lightly used, no markings",
                imgSelected: require("../../../img/condition_icons/Poor_condition.png")
            },
            {
                name: "Fair",
                img: require("../../../img/condition_icons/Fair_condition_default.png"),
                desc: "No damage, lightly used, no markings",
                imgSelected: require("../../../img/condition_icons/Fair_condition.png")
            },
            {
                name: "Good",
                img: require("../../../img/condition_icons/Good_condition_default.png"),
                desc: "No damage, lightly used, no markings",
                imgSelected: require("../../../img/condition_icons/Good_condition.png")
            },
            {
                name: "Very Good",
                img: require("../../../img/condition_icons/Verygood_condition_default.png"),
                desc: "No damage, lightly used, no markings",
                imgSelected: require("../../../img/condition_icons/Verygood_condition.png")
            },
            {
                name: "Like New",
                img: require("../../../img/condition_icons/Likenew_condition_default.png"),
                desc: "No damage, lightly used, no markings",
                imgSelected: require("../../../img/condition_icons/Likenew_condition.png")
            },
        ];

        let condEl = [];
        // eslint-disable-next-line
        conditions.map((cond, i) => {
            condEl.push(
                <div key={i} className="condition">
                    <p className="condition-title">{cond.name}</p>
                    <img className="condition-img"
                        src={this.state.condition === cond.name ? cond.imgSelected : cond.img}
                        alt={cond.name} name="condition" cond={cond.name}
                        onClick={() => this.selectCondition(cond.name)} />
                    <p className="condition-desc">{cond.desc}</p>
                </div>
            )
        });

        return (
            <div className="post-book-form">
                <div className="main-form">
                    <div className="post-book-img">
                        <img className="book-img" src="" alt="" />
                        <label htmlFor="upload-img" id="upload-btn">Upload Image</label>
                        <input id="upload-img" type="file" />
                    </div>

                    <form className="post-book-form-fields">
                        {form}
                    </form>
                </div>

                {editOpt}

                <div id="book-condition">
                    <p>Condition</p>
                    {condEl}
                </div>

                <button onClick={this.toggleEditable}>Save Changes</button>
            </div>
        );
    }
}

function mapStateToProps(state) {
    return {};
}

export default connect(
    mapStateToProps,
)(PostBookForm);
