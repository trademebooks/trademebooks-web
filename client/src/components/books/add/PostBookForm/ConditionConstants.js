import Poor_condition_default from '../../images/conditions/Poor_condition_default.png'
import Poor_condition from '../../images/conditions/Poor_condition.png'
import Fair_condition_default from '../../images/conditions/Fair_condition_default.png'
import Fair_condition from '../../images/conditions/Fair_condition.png'
import Good_condition_default from '../../images/conditions/Good_condition_default.png'
import Good_condition from '../../images/conditions/Good_condition.png'
import Verygood_condition_default from '../../images/conditions/Verygood_condition_default.png'
import Verygood_condition from '../../images/conditions/Verygood_condition.png'
import Likenew_condition_default from '../../images/conditions/Likenew_condition_default.png'
import Likenew_condition from '../../images/conditions/Likenew_condition.png'

export const CONDITION_TYPES = {
  POOR: 'POOR',
  FAIR: 'FAIR',
  GOOD: 'GOOD',
  VERY_GOOD: 'VERY_GOOD',
  LIKE_NEW: 'LIKE_NEW'
}

export default [
  {
    text: 'POOR',
    type: CONDITION_TYPES.POOR,
    description: 'Heavily worn, all text are still legible.',
    imageUrl: Poor_condition_default,
    imgSelected: Poor_condition
  },
  {
    text: 'FAIR',
    type: CONDITION_TYPES.FAIR,
    description: 'Lots of wear and tear.',
    imageUrl: Fair_condition_default,
    imgSelected: Fair_condition
  },
  {
    text: 'GOOD',
    type: CONDITION_TYPES.GOOD,
    description: 'Some signs of wear, includes markings.',
    imageUrl: Good_condition_default,
    imgSelected: Good_condition
  },
  {
    text: 'VERY GOOD',
    type: CONDITION_TYPES.VERY_GOOD,
    description: 'Minimal signs of wear, no markings.',
    imageUrl: Verygood_condition_default,
    imgSelected: Verygood_condition
  },
  {
    text: 'LIKE NEW',
    type: CONDITION_TYPES.LIKE_NEW,
    description: 'No damage, lightly used, no markings.',
    imageUrl: Likenew_condition_default,
    imgSelected: Likenew_condition
  }
]
