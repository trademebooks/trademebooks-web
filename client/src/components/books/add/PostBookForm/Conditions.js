import React, { useState } from 'react';
import { MDBRow, MDBCol } from 'mdbreact';

import './Conditions.scss';

const Conditions = () => {

  const [condition, setCondition] = useState('');

  let conditions = [
    {
      name: "Poor",
      img: require('../../../../img/condition_icons/Poor_condition_default.png'),
      desc: "No damage, lightly used, no markings",
      imgSelected: require("../../../../img/condition_icons/Poor_condition.png")
    },
    {
      name: "Fair",
      img: require("../../../../img/condition_icons/Fair_condition_default.png"),
      desc: "No damage, lightly used, no markings",
      imgSelected: require("../../../../img/condition_icons/Fair_condition.png")
    },
    {
      name: "Good",
      img: require("../../../../img/condition_icons/Good_condition_default.png"),
      desc: "No damage, lightly used, no markings",
      imgSelected: require("../../../../img/condition_icons/Good_condition.png")
    },
    {
      name: "Very Good",
      img: require("../../../../img/condition_icons/Verygood_condition_default.png"),
      desc: "No damage, lightly used, no markings",
      imgSelected: require("../../../../img/condition_icons/Verygood_condition.png")
    },
    {
      name: "Like New",
      img: require("../../../../img/condition_icons/Likenew_condition_default.png"),
      desc: "No damage, lightly used, no markings",
      imgSelected: require("../../../../img/condition_icons/Likenew_condition.png")
    },
  ];

  let conditionsElement = [];

  // eslint-disable-next-line
  conditions.map((cond, i) => {
    conditionsElement.push(
      <MDBCol md="2" sm="12" key={i} className="condition">
        <p className="condition-title">{cond.name}</p>
        <img className="condition-img"
          src={condition === cond.name ? cond.imgSelected : cond.img}
          alt={cond.name} name="condition" cond={cond.name}
          onClick={() => setCondition(cond.name)} />
        <p className="condition-desc">{cond.desc}</p>
      </MDBCol>
    )
  });

  return (
    <>
      <div className="mt-5">
        <MDBRow>
          {conditionsElement}
        </MDBRow>
      </div>
    </>
  );
}

export default Conditions;