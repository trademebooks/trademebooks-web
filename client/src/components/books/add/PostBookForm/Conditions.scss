.condition-img {
  width: 8rem;
  margin: 1rem 0rem;
  cursor: pointer;
}