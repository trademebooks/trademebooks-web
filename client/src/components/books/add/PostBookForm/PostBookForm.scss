.post-book-form-container {
  border: solid 2px black;
  margin: 1.5rem 0 1.5rem 0;
}