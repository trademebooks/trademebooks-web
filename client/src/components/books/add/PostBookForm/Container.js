import { connect } from 'react-redux';
import PostBookForm from './PostBookForm';

const mapStateToProps = (state) => ({
  book: state.book.book
});

const mapDispatchToProps = {

}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PostBookForm);
