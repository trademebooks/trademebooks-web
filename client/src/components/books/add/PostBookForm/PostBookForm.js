import React, { useEffect } from 'react';
import { MDBRow, MDBCol, MDBContainer, MDBInput, MDBBtn } from 'mdbreact';

import Conditions from './Conditions';

import './PostBookForm.scss';

const PostBookForm = ({ book }) => {
  useEffect(() => {
    console.log('BookFields', { book });
  }, [book]);

  return (
    <>
      <div className="post-book-form-container">
        <MDBContainer>
          <MDBRow>
            <MDBCol md="4" sm="12">
              <div>
                <div className="my-5">
                  <img
                    src="https://mdbootstrap.com/img/Others/documentation/img%20(75)-mini.jpg"
                    alt="thumbnail"
                    className="img-thumbnail"
                  />
                </div>
                <div className="custom-file">
                  <input
                    type="file"
                    className="custom-file-input"
                    id="inputGroupFile01"
                    aria-describedby="inputGroupFileAddon01"
                  />
                  <label
                    className="custom-file-label"
                    htmlFor="inputGroupFile01"
                  >
                    Choose file
                  </label>
                </div>
              </div>
            </MDBCol>
            <MDBCol md="8" sm="12">
              <div>
                <form className="post-book-form-fields">
                  <div className="form-group">
                    <MDBInput label="Title" size="lg" value={book.title} />
                  </div>
                  <div className="form-group">
                    <MDBInput label="ISBN-10" size="lg" />
                  </div>
                  <div className="form-group">
                    <MDBInput label="ISBN-13" size="lg" />
                  </div>
                  <div className="form-group">
                    <MDBInput label="Edition" size="lg" />
                  </div>
                </form>
              </div>

              <div>
                <MDBBtn>Save Changes</MDBBtn>
              </div>
            </MDBCol>
          </MDBRow>
          <MDBRow>
            <MDBCol md="12" sm="12">
              <Conditions />
            </MDBCol>
          </MDBRow>
        </MDBContainer>
      </div>
    </>
  );
};

export default PostBookForm;