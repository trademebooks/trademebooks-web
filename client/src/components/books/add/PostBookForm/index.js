import Container from './Container'; 

const PostBookForm = Container;

export default PostBookForm;
