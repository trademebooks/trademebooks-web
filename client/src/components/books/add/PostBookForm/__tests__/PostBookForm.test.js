import React from 'react'
import renderer from 'react-test-renderer'

import PostBookForm from '../PostBookForm'

describe('Component: PostBookForm', () => {
  const props = {
    book: jest.fn(),
    addBook: jest.fn()
  }

  it('should match the snapshot', () => {
    const PostBookFormSnapshot = renderer
      .create(<PostBookForm {...props} />)
      .toJSON()

    expect(PostBookFormSnapshot).toMatchSnapshot()
  })
})
