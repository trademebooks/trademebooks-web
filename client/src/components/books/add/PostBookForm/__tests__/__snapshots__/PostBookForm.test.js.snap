// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PostBookForm should match the snapshot 1`] = `
<div
  className="post-book-form-container"
>
  <div
    className="container"
    data-test="container"
  >
    <div
      className="row"
      data-test="row"
    >
      <div
        className="col-sm-12 col-md-4"
        data-test="col"
      >
        <div
          className="text-center"
        >
          <div
            className="py-5"
          >
            <img
              alt="thumbnail"
              className="img-thumbnail book-image-size"
              src="sample-book.png"
            />
          </div>
        </div>
      </div>
      <div
        className="col-sm-12 col-md-8"
        data-test="col"
      >
        <div>
          <form
            className="post-book-form-fields"
          >
            <div
              className="form-group"
            >
              <div
                className="md-form form-lg"
              >
                
                <input
                  aria-disabled={false}
                  className="form-control form-control-lg"
                  data-test="input"
                  disabled={false}
                  name="title"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onInput={[Function]}
                  type="text"
                  value=""
                />
                <label
                  aria-labelledby=""
                  className=""
                  data-error=""
                  data-success=""
                  id=""
                  onClick={[Function]}
                >
                  Title (Required)
                </label>
              </div>
            </div>
            <div
              className="form-group"
            >
              <div
                className="md-form form-lg"
              >
                
                <input
                  aria-disabled={false}
                  className="form-control form-control-lg"
                  data-test="input"
                  disabled={false}
                  name="edition"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onInput={[Function]}
                  type="text"
                  value=""
                />
                <label
                  aria-labelledby=""
                  className=""
                  data-error=""
                  data-success=""
                  id=""
                  onClick={[Function]}
                >
                  Edition (Optional)
                </label>
              </div>
            </div>
            <div
              className="form-group"
            >
              <div
                className="tags-input"
              >
                <ul
                  id="tags"
                />
                <input
                  onKeyUp={[Function]}
                  placeholder="Author Name(s) (Optional)"
                  type="text"
                />
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
    <div
      className="row"
      data-test="row"
    >
      <div
        className="col-sm-12 col-md-12"
        data-test="col"
      >
        <div
          className="mt-5"
        >
          <div
            className="row justify-content-center"
            data-test="row"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
